docs/DeveloperGuide.adoc

```diff
@@ -153,9 +154,138 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+=== Global Features
+
+// tag::tabswitching[]
+==== Separate Parsing and Tab Switching
+===== Implementation
+===== Design Considerations
+// end::tabswitching[]
+
+// tag::tasks[]
+==== Tasks
+Some models in IchiFund must be refreshed after a command is executed.
+For instance, when a new `Transaction` is added, all `Budget` must be recomputed.
+`Task` can be used to facilitate such updates.
+
+===== Implementation
+This feature is managed by `TaskManager`.
+The role of `TaskManager` is to maintain a list of all active `Task`.
+
+The `LogicManager` holds an instance of the `TaskManager`.
+When the `LogicManager#execute()` is called, the following chain of operations occurs:
+
+1. After `Command#execute()` is completed, `TaskManager#executeAll()` is called.
+
+image::TaskCode.png[]
```

Comment:
An example usage scenario could be useful in making the purpose of Tasks clearer.